### PR TITLE
LPS-171385 Add a view to handle the collection configuration edition

### PR DESCRIPTION
**Pull request by lfbesada**
https://issues.liferay.com/browse/LPS-171385

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by lfbesada** [comment]
ci:test:sf

**Comment by lfbesada** [comment]
ci:test:relevant

**Comment by lfbesada** [comment]
ci:test:sf

**Comment by ealonso** [comment]
Just started reviewing :)

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/EditCollectionConfigurationMVCRenderCommand.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	@Override
+	public String render(
+		RenderRequest renderRequest, RenderResponse renderResponse) {
+
+		HttpServletRequest httpServletRequest = _portal.getHttpServletRequest(
+			renderRequest);
+
+		EditCollectionConfigurationDisplayContext
+			editCollectionConfigurationDisplayContext =
+				new EditCollectionConfigurationDisplayContext(
```
Comment (ealonso) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/EditCollectionConfigurationMVCRenderCommand.java:54]: @lfbesada you can inline it

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/EditCollectionConfigurationMVCRenderCommand.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	service = MVCRenderCommand.class
+)
+public class EditCollectionConfigurationMVCRenderCommand
+	implements MVCRenderCommand {
+
+	@Override
+	public String render(
+		RenderRequest renderRequest, RenderResponse renderResponse) {
+
+		HttpServletRequest httpServletRequest = _portal.getHttpServletRequest(
```
Comment (ealonso) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/EditCollectionConfigurationMVCRenderCommand.java:49]: @lfbesada also inline this one

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/GetEditCollectionConfigurationURLMVCResourceCommand.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		String collectionKey = ParamUtil.getString(
+			resourceRequest, "collectionKey");
+
+		if (Validator.isBlank(collectionKey)) {
```
Comment (ealonso) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/GetEditCollectionConfigurationURLMVCResourceCommand.java:64]: @lfbesada usually we use isNull

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/GetEditCollectionConfigurationURLMVCResourceCommand.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+			ParamUtil.getString(
+				_portal.getOriginalServletRequest(
+					_portal.getHttpServletRequest(resourceRequest)),
+				"urlCurrent"),
+			"itemId");
+
+		urlCurrent = HttpComponentsUtil.addParameter(
+			urlCurrent, "itemId", itemId);
+
+		JSONObject jsonObject = JSONUtil.put(
```
Comment (ealonso) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/GetEditCollectionConfigurationURLMVCResourceCommand.java:104]: @lfbesada you can inline it

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/edit_collection_configuration.jsp [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+EditCollectionConfigurationDisplayContext editCollectionConfigurationDisplayContext = (EditCollectionConfigurationDisplayContext)request.getAttribute(EditCollectionConfigurationDisplayContext.class.getName());
+
+portletDisplay.setShowBackIcon(true);
+portletDisplay.setURLBack(editCollectionConfigurationDisplayContext.getRedirect());
+
+renderResponse.setTitle(LanguageUtil.get(request, "filter-collection"));
+%>
+
+<liferay-frontend:edit-form
+	action="<%= editCollectionConfigurationDisplayContext.getActionURL() %>"
```
Comment (ealonso) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/edit_collection_configuration.jsp:29]: @lfbesada Why not using the taglibs to create the action url?

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/UpdateCollectionConfigurationMVCActionCommand.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+				});
+
+			sendRedirect(
+				actionRequest, actionResponse,
+				ParamUtil.getString(actionRequest, "redirect"));
+		}
+		catch (Exception exception) {
+			_log.error(exception);
+
+			hideDefaultErrorMessage(actionRequest);
```
Comment (ealonso) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/UpdateCollectionConfigurationMVCActionCommand.java:111]: @lfbesada Why you are hiding the default error? It seems to be the same that you are adding

**Comment by ealonso** [comment]
see: https://github.com/liferay-echo/liferay-portal/pull/10724

**Pull request closed without merging**